### PR TITLE
Fix IndexError in plot_fractal_tree

### PR DESCRIPTION
**Pull request by Hydrogenesi**
`The` `plot_fractal_tree` function could raise an `IndexError` if a proton with an empty lineage was present in the `proton_lineage` list. This was because the list comprehension used to find children of a proton accessed `p.lineage[-1]` without first checking if `p.lineage` was empty.

The fix adds a check to ensure `p.lineage` is not empty before accessing its last element.

A new test case was added in `test_plotting.py` to specifically target this edge case and verify the fix. The test fails before the fix and passes after.

### CODE REVIEW

File: test_plotting.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+class TestPlotting(unittest.TestCase):
+
+    def test_plot_fractal_tree_index_error(self):
+        """
+        Tests that the plot_fractal_tree function does not raise an IndexError
+        when a proton with an empty lineage is present.
+        """
+        fig, ax = plt.subplots()
+        initial_proton = Proton("P0", [])
+        proton_lineage.append(initial_proton)
```
[the remark this answers is not in the document]
Comment (Hydrogenesi) [reply test_plotting.py:17]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/Hydrogenesi/TOE/pull/4#r2525660959)

File: test_plotting.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+        try:
+            plot_fractal_tree(ax, initial_proton, 0, 0, 90, 1, 1)
+        except IndexError:
+            self.fail("plot_fractal_tree() raised IndexError unexpectedly!")
```
[the remark this answers is not in the document]
Comment (Hydrogenesi) [reply test_plotting.py]: def test_plot_fractal_tree_index_error(self):
    """
    Tests that the plot_fractal_tree function does not raise an IndexError
    when a proton with an empty lineage is present.
    """
    fig, ax = plt.subplots()
    initial_proton = Proton("P0", [])
    proton_lineage.append(initial_proton)
    
    # This will fail the test automatically if IndexError is raised
    plot_fractal_tree(ax, initial_proton, 0, 0, 90, 1, 1)

**Review by Hydrogenesi — COMMENTED** [review]
JPS

**Comment by Hydrogenesi** [comment]
@copilot open a new pull request to apply changes based on the comments in [this thread](https://github.com/Hydrogenesi/TOE/pull/4#pullrequestreview-3462668782)

**Review by Hydrogenesi — COMMENTED** [review]
JPS

**Review by Hydrogenesi — COMMENTED** [review]
What's the exact problem is it transition or ?

**Review by Hydrogenesi — COMMENTED** [review]
JPS

**Comment by Hydrogenesi** [comment]
@copilot open a new pull request to apply changes based on the comments in [this thread](https://github.com/Hydrogenesi/TOE/pull/4#pullrequestreview-3462668782)

**Comment by Hydrogenesi** [comment]
@copilot open a new pull request to apply changes based on the comments in [this thread](https://github.com/Hydrogenesi/TOE/pull/4#pullrequestreview-3462668782)

**Comment by Hydrogenesi** [comment]
JPS
